src-gui/src/renderer/components/pages/swap/swap/in_progress/offer/Offer.tsx

```diff
@@ -0,0 +1,162 @@
+import {
+  DialogContent,
+  DialogActions,
+  Typography,
+  Box,
+  Divider,
+} from "@mui/material";
+import {
+  PendingLockBitcoinApprovalRequest,
+  TauriSwapProgressEventContent,
+} from "models/tauriModelExt";
+import { useEffect, useState } from "react";
+import { useActiveSwapId, usePendingLockBitcoinApproval } from "store/hooks";
+import { resolveApproval } from "renderer/rpc";
+import { SatsAmount } from "renderer/components/other/Units";
+import CircularProgressWithSubtitle from "renderer/components/pages/swap/swap/components/CircularProgressWithSubtitle";
+import PromiseInvokeButton from "renderer/components/PromiseInvokeButton";
+import { Check as CheckIcon } from "@mui/icons-material";
+import CancelButton from "../../CancelButton";
+
+function useActiveLockBitcoinApprovalRequest(): PendingLockBitcoinApprovalRequest | null {
+  const approvals = usePendingLockBitcoinApproval();
+  const activeSwapId = useActiveSwapId();
+
+  return (
+    approvals?.find(
+      (r) => r.content.details.content.swap_id === activeSwapId,
+    ) || null
+  );
+}
+
+export default function Offer({
+  btc_lock_amount,
+}: TauriSwapProgressEventContent<"SwapSetupInflight">) {
+  const request = useActiveLockBitcoinApprovalRequest();
+
+  const [timeLeft, setTimeLeft] = useState<number>(0);
+
+  const expiresAtMs = request?.content.expiration_ts * 1000 || 0;
+
+  useEffect(() => {
+    const tick = () => {
+      const remainingMs = Math.max(expiresAtMs - Date.now(), 0);
+      setTimeLeft(Math.ceil(remainingMs / 1000));
+    };
+
+    tick();
+    const id = setInterval(tick, 250);
+    return () => clearInterval(id);
+  }, [expiresAtMs]);
+
+  if (!request) {
+    return (
+      <>
+        <CircularProgressWithSubtitle
+          description={
+            <>
+              Negotiating offer for <SatsAmount amount={btc_lock_amount} />
+            </>
+          }
+        />
+        <CancelButton />
+      </>
+    );
+  }
+
+  const { btc_network_fee, xmr_receive_amount } =
+    request.content.details.content;
+
+  return (
+    <>
+      <Box>
```

Comment:
This whole component was changed on master.

Please try to merge master into this branch and use the new offer page layout.